test/server/admin/stats_handler_test.cc

```diff
@@ -15,45 +15,82 @@ using testing::EndsWith;
 using testing::HasSubstr;
 using testing::InSequence;
 using testing::Ref;
+using testing::Return;
+using testing::ReturnRef;
 using testing::StartsWith;
 
 namespace Envoy {
 namespace Server {
 
 class StatsHandlerTest {
 public:
-  StatsHandlerTest() : alloc_(symbol_table_), pool_(symbol_table_) {
+  StatsHandlerTest() : pool_(symbol_table_), alloc_(symbol_table_) {
     store_ = std::make_unique<Stats::ThreadLocalStoreImpl>(alloc_);
     store_->addSink(sink_);
+    store_->initializeThreading(main_thread_dispatcher_, tls_);
   }
 
-  static std::string
-  statsAsJsonHandler(std::map<std::string, uint64_t>& all_stats,
-                     std::map<std::string, std::string>& all_text_readouts,
-                     const std::vector<Stats::ParentHistogramSharedPtr>& all_histograms,
-                     const bool used_only, const absl::optional<std::regex> regex = absl::nullopt) {
-    return StatsHandler::statsAsJson(all_stats, all_text_readouts, all_histograms, used_only, regex,
-                                     true /*pretty_print*/);
-  }
-
-  Stats::StatName makeStat(absl::string_view name) { return pool_.add(name); }
-  Stats::CustomStatNamespaces& customNamespaces() { return custom_namespaces_; }
-
-  void shutdownThreading() {
+  ~StatsHandlerTest() {
     tls_.shutdownGlobalThreading();
     store_->shutdownThreading();
     tls_.shutdownThread();
   }
 
+  using CodeResponse = std::pair<Http::Code, std::string>;
+
+  /**
+   * Issues a request for stat as json.
+   *
+   * @param used_only only include stats that have been written.
+   * @param filter string interpreted as regex to filter stats
+   * @return the Http Code and the response body as a string.
+   */
+  CodeResponse statsAsJsonHandler(const bool used_only,
+                                  const absl::optional<std::string> filter = absl::nullopt) {
```

Comment:
We don't need const here since we're passing by value and will create a copy when calling